python/ray/tests/accelerators/test_amd_gpu.py

```diff
@@ -0,0 +1,87 @@
+import os
+import sys
+import pytest
+from unittest.mock import patch
+
+import ray
+from ray._private.accelerators import AMDGPUAcceleratorManager as Accelerator
+
+
+def test_visible_amd_gpu_ids(shutdown_only):
+    with patch.object(Accelerator, "get_current_node_num_accelerators", return_value=4):
+        os.environ["ROCR_VISIBLE_DEVICES"] = "0,1,2"
+        ray.init()
+        manager = ray._private.accelerators.get_accelerator_manager_for_resource("GPU")
+        assert manager.get_current_node_num_accelerators() == 4
+        assert manager.__name__ == "AMDGPUAcceleratorManager"
+        assert ray.available_resources()["GPU"] == 3
+
+
+def test_visible_amd_gpu_type(shutdown_only):
+    with patch.object(
+        Accelerator,
+        "_get_amd_pci_ids",
+        return_value={
+            "card0": {"GPU ID": "0x740f"},
+            "card1": {"GPU ID": "0x740f"},
+            "card2": {"GPU ID": "0x740f"},
+            "card3": {"GPU ID": "0x740f"},
+        },
+    ):
+        ray.init()
+        manager = ray._private.accelerators.get_accelerator_manager_for_resource("GPU")
+        assert manager.get_current_node_accelerator_type() == "AMD Instinct MI210"
+
+    with patch.object(
+        Accelerator, "_get_amd_pci_ids", return_value={"card0": {"GPU ID": "0x640f"}}
+    ):
+        manager = None
+        manager = ray._private.accelerators.get_accelerator_manager_for_resource("GPU")
+        assert manager.get_current_node_accelerator_type() is None
+
+    with patch.object(Accelerator, "_get_amd_pci_ids", return_value=None):
+        manager = None
+        manager = ray._private.accelerators.get_accelerator_manager_for_resource("GPU")
+        assert manager.get_current_node_accelerator_type() is None
+
+
+def test_amd_gpu_accelerator_manager_api():
+    assert Accelerator.get_resource_name() == "GPU"
+    assert Accelerator.get_visible_accelerator_ids_env_var() == "ROCR_VISIBLE_DEVICES"
+    assert Accelerator.validate_resource_request_quantity(0.1) == (True, None)
+
+
+def test_get_current_process_visible_accelerator_ids():
+    os.environ["ROCR_VISIBLE_DEVICES"] = "0,1,2"
```

Comment:
Use `monkeypatch` to set env var so it will automatically reset it after the test finishes. For example:

```
def test_healthz_agent_2(monkeypatch, ray_start_cluster):
    monkeypatch.setenv("RAY_health_check_failure_threshold", "3")
```

Change other tests to use it as well.